core/services/log/broadcaster_internal_test.go

```diff
@@ -0,0 +1,176 @@
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/flux_aggregator_wrapper"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils/pgtest"
+	ethmocks "github.com/smartcontractkit/chainlink/core/services/eth/mocks"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type tc struct {
+}
+
+func (tc) BlockBackfillDepth() uint64 {
+	return 0
+}
+func (tc) BlockBackfillSkip() bool {
+	return true
+}
+func (tc) EthFinalityDepth() uint {
+	return 1
+}
+func (tc) EthLogBackfillBatchSize() uint32 {
+	return 1
+}
+
+type listener struct {
+	logs chan Broadcast
+}
+
+func (l listener) HandleLog(b Broadcast) {
+	l.logs <- b
+}
+
+func (l listener) JobID() models.JobID {
+	return models.NewJobID()
+}
+
+func (l listener) JobIDV2() int32 {
+	return 1
+}
+
+func (l listener) IsV2Job() bool {
+	return true
+}
+
+type sub struct {
+}
+
+func (s sub) Unsubscribe() {
+}
+
+func (s sub) Err() <-chan error {
+	return nil
+}
+
+func TestBroadcaster_BroadcastsWithZeroConfirmations(t *testing.T) {
+	logsChCh := make(chan chan<- types.Log)
+	ec := new(ethmocks.Client)
+	ec.On("SubscribeFilterLogs", mock.Anything, mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			logsChCh <- args.Get(2).(chan<- types.Log)
+		}).
+		Return(sub{}, nil)
+	ec.On("HeadByNumber", mock.Anything, (*big.Int)(nil)).
+		Return(&models.Head{Number: 1}, nil)
+	ec.On("FilterLogs", mock.Anything, mock.Anything).
+		Return(nil, nil)
+	db := pgtest.NewGormDB(t)
+	dborm := NewORM(db)
+	lb := NewBroadcaster(dborm, ec, tc{}, nil)
+	lb.Start()
+	// TODO: make this not hang
+	//defer lb.Close()
+
+	addr := common.HexToAddress("0xf0d54349aDdcf704F77AE15b96510dEA15cb7952")
+	contract1, err := flux_aggregator_wrapper.NewFluxAggregator(addr, nil)
+	require.NoError(t, err)
+
+	bh := utils.NewHash()
+	addr1SentLogs := []types.Log{
+		{
+			Address:     addr,
+			BlockHash:   bh,
+			BlockNumber: 2,
+			Index:       0,
+			Topics: []common.Hash{
+				(flux_aggregator_wrapper.FluxAggregatorNewRound{}).Topic(),
+				utils.NewHash(),
+				utils.NewHash(),
+			},
+			Data: []byte("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+		},
+		{
+			Address:     addr,
+			BlockHash:   bh,
+			BlockNumber: 2,
+			Index:       1,
+			Topics: []common.Hash{
+				(flux_aggregator_wrapper.FluxAggregatorNewRound{}).Topic(),
+				utils.NewHash(),
+				utils.NewHash(),
+			},
+			Data: []byte("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+		},
+		{
+			Address:     addr,
+			BlockHash:   bh,
+			BlockNumber: 2,
+			Index:       2,
+			Topics: []common.Hash{
+				(flux_aggregator_wrapper.FluxAggregatorNewRound{}).Topic(),
+				utils.NewHash(),
+				utils.NewHash(),
+			},
+			Data: []byte("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+		},
+	}
```

Comment:
could use `cltest.RawNewRoundLog` for setting up these

Reply:
cltest creates an import cycle. I'm general I think it should only be used for integration level tests